include/Chibi.hpp

```diff
@@ -25,9 +26,45 @@ class Chibi {
 
         @arg name name, under which this function will be known in scheme
         @arg fnc the function pointer to use
+        @arg opcode_data optional data associated with the function opcode
      */
     template <typename... Args>
-    SExp register_function(const std::string &name, sexp (*fnc)(sexp, sexp, long, Args...));
+    SExp register_function(const std::string &name, sexp (*fnc)(sexp, sexp, long, Args...), sexp opcode_data = nullptr);
+
+    /** Class that helps adding C++ class members to scheme. */
+    template<typename Class>
+    class ClassRegistrator {
+    public:
+        /** Create an instance that will prepand \a name to it's members when registering. */
+        ClassRegistrator(Chibi &chibi, std::string &name);
+
+        /** Register a member function and add it to scheme under the name "<class-name>-<member-name>".
+         *
+         * The registered function will accept a cpointer with a pointer to the class used as `this` and all
+         * other arguments after that will be passed to the function. They'll be converted from sexp to C++
+         * using the #make_from function. The result will be converted from C++ back to scheme using the #SExp.to
+         * function.
+         */
+        template<typename Return, typename... Args>
+        ClassRegistrator<Class> &register_method(std::string &&name, Return (Class::*member_function)(Args...));
+
+        /** Register a class field to be accessible from scheme under the name "<class-name>-<field-name>".
+         *  Optionally, add a setter named "set-<class-name>-<field-name>!".
+         *
+         * @arg name field name
+         * @arg field field reference
+         * @arg generate_setter whether to generate a setter, it's not generated by default
+         */
+        template<typename FieldType>
+        ClassRegistrator<Class> &register_field(std::string &&name, FieldType Class::*field, bool generate_setter = false);
+    private:
+        Chibi &chibi;
+
+        std::string &class_name;
```

Comment:
Same as above. Also `std::string` is small enough to just copy it.
Please don't do optimizations on this stage, it just would backfire later, when things become really complex.